### ISSUE TITLE
[Possible Bug]: Serialized model using `TorchModuleWrapper` does not reproduce evaluation metrics

### ISSUE THREAD

**Issue by soumik12345** [keras-team/keras#18403]
I noticed that when a Keras model using `TorchModuleWrapper` is serialized and evaluated after training, it does not produce the same metrics (accuracy and loss) upon being evaluated. In fact, it seems that the weights are not being serialized properly.

Colab to reproduce issue: https://colab.research.google.com/drive/1IweLgXnyCNY9a2VJ2UkCqvntDurfLobN?usp=sharing

**Comment by sampathweb** [comment]
Looks like an issue with the usage of BatchNorm layers in the underlying torch model.  If you build a model without `BatchNorm` layers it works fine.  Still investigating in wip PR #18498

**Comment by sampathweb** [comment]
Here's a [minimal example](https://gist.github.com/sampathweb/b3105b4dc0e9cde65bfaf07d31abc14a
) of the issue.  Notice that `model.parameters()` doesn't get BatchNorm variables. 
This is documented in PyTorch as well that `BatchNorm` is not included in `model.parameters()`, but is trained and also part of `model.state_dict()`.

**Comment by sampathweb** [comment]
Closed with #18498
